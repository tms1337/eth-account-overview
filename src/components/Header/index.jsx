import React from "react";
import { createUseStyles } from "react-jss";

import { colors } from "../../config/ui";

const Header = ({ title, logo }) => {
  const styles = useStyles();

  return (
    <div className={styles.root}>
      {logo && <img src={logo} alt="Company logo" />}
      <h2>{title}</h2>
    </div>
  );
};

const useStyles = createUseStyles({
  root: {
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    background: colors.PRIMARY,
    color: colors.WHITE
  }
});

export default Header;
