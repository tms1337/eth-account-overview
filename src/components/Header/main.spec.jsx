import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Header from ".";

describe("Header", () => {
  it("renders snapshot", () => {
    const component = renderer.create(<Header />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("displays input with title", () => {
    const title = "Enter your wallet address";
    const component = mount(<Header title={title} />);

    expect(component.find("h2").length).toBe(1);
    expect(component.find("h2").text()).toBe(title);
  });
});
