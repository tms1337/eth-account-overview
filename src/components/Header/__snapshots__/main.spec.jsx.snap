// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders snapshot 1`] = `
<div
  className="root-0-2-1"
>
  <h2 />
</div>
`;
